### DISCUSSION TITLE
Pointer to the current page

### DISCUSSION THREAD

**Discussion by alex13sh** [slint-ui/slint#2275] [Q&A]
Why can't I use the same name in **Conditional element**?

Or how can I create multiple pages that I can switch between and specify the current page? I want to use the title property of a component when I have a pointer to the open page. How can I do this?

In the `gallery` example, there are several pages, and the **`Conditional element`** is used to switch between them. How can I use the title property from the Page component for Window.title in this example?

**Comment by tronical** [comment]
I'm not sure if there's a great way for this. Once #2344 is resolved, the one thing that you can do is set the outer property from with an init callback, like this:

(from the gallery.slint file)
```
        if(side-bar.current-item == 0) : ControlsPage {
            init() => {
                root.title = self.title;
            }
        }
```

**Reply by alex13sh** [reply]
Is it not possible to use a pointer on the base component? Or could this be possible in the future? For example, in Qt it is possible to store a pointer to a base component like this:
```
property BaseComponent base_pointer: FooComponent {}
property string title: base_pointer.title
```
And when the pointer changes to another component, the title property is updated automatically. Is it currently not possible to have such behavior?

I would like something like this.
```
component Base {
    in property <string> title: "Base Component";
}

component Foo inherits Base {
    title: "Foo Component";
}

component Bar inherits Base {
    title: "Bar Component";
}

export component TestCase inherits Rectangle {
    width: 300phx;
    height: 300phx;

    in property <Base> base: Foo {};
    
    Text {
        text <=> base.title;
    }

    in property <bool> cond: false;

    if cond: root.base = Bar {}

    if !cond: root.base = Foo {}
}
``
